src/statsd_client.cpp

```diff
@@ -1,15 +1,42 @@
-
 #include <math.h>
-#include <netdb.h>
 #include <time.h>
-#include <unistd.h>
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
 #include <sys/types.h>
-#include <netinet/in.h>
 #include "statsd_client.h"
-#include <fcntl.h>
+
+/* platform-specific headers */
+#ifdef HAVE_WINSOCK2_H
```

Comment:
I'd be tempted to put all the conditional checks in here directly. Reasoning being that we essentially have two modes: POSIX and Win32. If we only have some of the headers for one or the other, things will break anyway.

```cpp
#ifdef POSIX
#include <posix things>
#elsif WIN32
#include <win32 things>
#else
#error "Need either POSIX or WIN32, this platform seems to be neither"
#endif
```

that way all the checks can be removed from cmake+the config.h file.